src/tiny.ts

```diff
@@ -28,22 +28,32 @@ export class TinyPg {
    private transaction_id?: string
 
    constructor(options: T.TinyPgOptions) {
+      options = _.isNil(options) ? {} : options
+
       this.events = new EventEmitter()
       this.error_transformer = _.isFunction(options.error_transformer) ? options.error_transformer : _.identity
       this.options = options
       this.hooks = _.isNil(this.options.hooks) ? [] : [this.options.hooks]
 
-      const params = Url.parse(options.connection_string, true)
-      const [user, password] = _.isNil(params.auth) ? ['postgres', undefined] : params.auth.split(':', 2)
+      const default_user = _.isNil(process.env.PGUSER) ? 'postgres' : process.env.PGUSER
+      const default_password = _.isNil(process.env.PGPASSWORD) ? undefined : process.env.PGPASSWORD
+      const default_host = _.isNil(process.env.PGHOST) ? 'localhost' : process.env.PGHOST
+      const default_database = _.isNil(process.env.PGDATABASE) ? 'postgres' : process.env.PGDATABASE
+      const default_port = _.isNil(process.env.PGPORT) ? 5432 : _.toInteger(process.env.PGPORT)
+      const default_ssl = _.isNil(process.env.PGSSLMODE) ? 'disable' : process.env.PGSSLMODE
+
+      const params = Url.parse(_.isNil(options.connection_string) ? '' : options.connection_string, true)
+      const [user, password] = _.isNil(params.auth) ? [default_user, default_password] : params.auth.split(':', 2)
       const pool_options = _.isNil(options.pool_options) ? {} : options.pool_options
-      const port = _.isNil(params.port) ? 5432 : _.toInteger(params.port)
-      const database = _.isNil(params.pathname) ? 'localhost' : params.pathname.split('/')[1]
-      const enable_ssl = _.get(params.query, 'sslmode') !== 'disable'
+      const port = _.isNil(params.port) ? default_port : _.toInteger(params.port)
+      const database = _.isNil(params.pathname) ? default_database : params.pathname.split('/')[1]
+      const enable_ssl = !_.includes(['disable', 'allow'], _.get(params.query, 'sslmode', default_ssl))
```

Comment:
considering some of this logic it may be useful to create a connection options function that can be tested separately.